tests/unit/db/view_execution_tests.py

```diff
@@ -0,0 +1,602 @@
+        ...
+        """
+        actual = self.view005(endkey=['julia', 9])['rows']
+        expected = [{'key': ['julia', x],
+                     'id': 'julia{0:03d}'.format(x),
+                     'value': 1} for x in range(10)]
+        self.assertEqual(len(actual), 10)
+        self.assertEqual(len(expected), 10)
+        self.assertEqual(actual, expected)
+
+    def test_endkey_docid(self):
+        """
+        Test view query using endkey_docid parameter.
+
+        The view used here will generate rows of data where each key will have
+        two ids associated with it.  Such as:
+        {'key': 0, 'id': 'julia000', 'value': 1},
+        {'key': 0, 'id': 'julia001', 'value': 1},
+        {'key': 1, 'id': 'julia002', 'value': 1},
+        {'key': 1, 'id': 'julia003', 'value': 1},
+        ...
+        {'key': 5, 'id': 'julia010', 'value': 1},
+        {'key': 5, 'id': 'julia011', 'value': 1},
+        ...
+        """
+        # Ensure that only rows of data up to and including the first document 
+        # where the key is 5 are returned.
+        actual = self.view003(endkey_docid='julia010', endkey=5)['rows']
+        expected = [{'key': x // 2,
+                     'id': 'julia{0:03d}'.format(x),
+                     'value': 1} for x in range(11)]
+        self.assertEqual(len(actual), 11)
+        self.assertEqual(len(expected), 11)
+        self.assertEqual(actual, expected)
+
+    def test_group_true(self):
+        """
+        Test view query using group parameter set to True.
+
+        The view used here along with group=True will generate rows of
+        data where each key will be grouped into groups of 2.  Such as:
+        {'key': 0, 'value': 2},
+        {'key': 1, 'value': 2},
+        {'key': 2, 'value': 2},
+        ...
+        {'key': 49, 'value': 2}
+        """
+        actual = self.view004(group=True)['rows']
+        expected = [{'key': x, 'value': 2} for x in range(50)]
+        self.assertEqual(len(actual), 50)
+        self.assertEqual(len(expected), 50)
+        self.assertEqual(actual, expected)
+
+    def test_group_false(self):
+        """
+        Test view query using group parameter set to False.
+
+        The view used here will generate a row of data containing the number of
+        documents matching the view query.  Such as:
+        {'key': None, 'value': 100}
+        """
+        actual = self.view004(group=False)['rows']
+        self.assertEqual(actual, [{'key': None, 'value': 100}])
+
+    def test_group_level(self):
+        """
+        Test view query using group_level parameter.
+
+        The view used here along with group_level=1 will generate rows of
+        data where each key will equal the document id.  Such as:
+        {'key': 'julia000', 'value': 1},
+        {'key': 'julia001', 'value': 1},
+        {'key': 'julia002', 'value': 1},
+        ...
+        """
+        actual = self.view002(group_level=1)['rows']
```

Comment:
I see your point that using view002 is not a very compelling example and it probably does not test the true nature of the `group_level` parameter.  view005 would not work since it does not contain a reduce function.  I added a view006 which is a copy of view005 with a reduce to `_count` and changed this test to use that view with `group_level=1`.  Fixed in https://github.com/cloudant/python-cloudant/pull/119/commits/9b90772fadf62077492db923a95ee032332b568d.